invenio_drafts_resources/services/records/components/base.py

```diff
@@ -251,9 +255,11 @@ def delete_draft(self, identity, draft=None, record=None, force=False):
             deleted instead of soft deleted (i.e. an unpublished draft).
         """
         draft_files = self.get_record_files(draft)
-        draft_files.unlock()
+        files_locked = self.service.config.lock_edit_published_files(record)
```

Comment:
Comment: Is this needed......shouldn't delete draft just wipe out the draft bucket independently on if it's locked or not.......it's only the need to call ``lock_edit_published_files()`` that seems off to me.